auth/service.go

```diff
@@ -997,70 +848,44 @@ func DecodeDomainUserID(domainUserID string) (string, string) {
 	}
 }
 
-func (svc service) DeleteEntityPolicies(ctx context.Context, entityType, id string) (err error) {
-	switch entityType {
-	case ThingType:
-		req := PolicyReq{
-			Object:     id,
-			ObjectType: ThingType,
-		}
-
-		return svc.DeletePolicyFilter(ctx, req)
-	case UserType:
-		domainsPage, err := svc.domains.ListDomains(ctx, Page{SubjectID: id, Limit: defLimit})
-		if err != nil {
-			return err
-		}
-
-		if domainsPage.Total > defLimit {
-			for i := defLimit; i < int(domainsPage.Total); i += defLimit {
-				page := Page{SubjectID: id, Offset: uint64(i), Limit: defLimit}
-				dp, err := svc.domains.ListDomains(ctx, page)
-				if err != nil {
-					return err
-				}
-				domainsPage.Domains = append(domainsPage.Domains, dp.Domains...)
-			}
-		}
+func (svc service) DeleteUserPolicies(ctx context.Context, id string) (err error) {
+	domainsPage, err := svc.domains.ListDomains(ctx, Page{SubjectID: id, Limit: defLimit})
+	if err != nil {
+		return err
+	}
 
-		for _, domain := range domainsPage.Domains {
-			policy := PolicyReq{
-				Subject:     EncodeDomainUserID(domain.ID, id),
-				SubjectType: UserType,
-			}
-			if err := svc.agent.DeletePolicyFilter(ctx, policy); err != nil {
+	if domainsPage.Total > defLimit {
+		for i := defLimit; i < int(domainsPage.Total); i += defLimit {
+			page := Page{SubjectID: id, Offset: uint64(i), Limit: defLimit}
+			dp, err := svc.domains.ListDomains(ctx, page)
+			if err != nil {
 				return err
 			}
+			domainsPage.Domains = append(domainsPage.Domains, dp.Domains...)
 		}
+	}
 
-		req := PolicyReq{
-			Subject:     id,
+	for _, domain := range domainsPage.Domains {
+		req := policies.PolicyReq{
+			Subject:     EncodeDomainUserID(domain.ID, id),
 			SubjectType: UserType,
 		}
-		if err := svc.agent.DeletePolicyFilter(ctx, req); err != nil {
-			return err
-		}
-
-		if err := svc.domains.DeleteUserPolicies(ctx, id); err != nil {
+		if err := svc.policies.DeletePolicyFilter(ctx, req); err != nil {
 			return err
 		}
+	}
 
-		return nil
-	case GroupType:
-		req := PolicyReq{
-			SubjectType: GroupType,
-			Subject:     id,
-		}
-		if err := svc.DeletePolicyFilter(ctx, req); err != nil {
-			return err
-		}
+	req := policies.PolicyReq{
+		Subject:     id,
+		SubjectType: UserType,
+	}
+	if err := svc.policies.DeletePolicyFilter(ctx, req); err != nil {
+		return err
+	}
```

Comment:
This block can be moved to `delete_handler.go` in users service at here https://github.com/absmach/magistrala/blob/main/users/delete_handler.go#L90-L91